src/commands/mod.rs

```diff
@@ -188,6 +194,22 @@ impl DaemonControl {
         };
         desc.derive(coin.derivation_index, &self.secp)
     }
+
+    // Check whether this address is valid for the network we are operating on.
+    fn validate_address(&self, addr: &bitcoin::Address) -> Result<(), CommandError> {
+        // NOTE: signet uses testnet addresses
+        if addr.network == self.config.bitcoin_config.network
+            || (addr.network == bitcoin::Network::Testnet
+                && self.config.bitcoin_config.network == bitcoin::Network::Signet)
```

Comment:
does not handle regtest.

Reply:
Regtest has its own address prefix, `bcrt`.